src/torchcodec/samplers/_implem.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+
+import torch
+
+from torchcodec.decoders import (  # TODO: move FrameBatch to torchcodec.FrameBatch?
+    FrameBatch,
+    SimpleVideoDecoder,
+)
+
+
+def clips_at_random_indices(
+    decoder: SimpleVideoDecoder,
+    *,
+    num_clips: int = 1,
+    num_frames_per_clip: int = 1,
+    num_indices_between_frames: int = 1,
+) -> List[FrameBatch]:
+    if num_clips <= 0:
+        raise ValueError(f"num_clips ({num_clips}) must be strictly positive")
+    if num_frames_per_clip <= 0:
+        raise ValueError(
+            f"num_frames_per_clip ({num_frames_per_clip}) must be strictly positive"
+        )
+    if num_indices_between_frames <= 0:
+        raise ValueError(
+            f"num_indices_between_frames ({num_indices_between_frames}) must be strictly positive"
+        )
```

Comment:
Is it possible for our decoder to have length 0? That is, we have a video file with metadata and no frames? If yes, we should probably check that here. That is the one condition where it will not be possible for us to respect `num_frames_per_clip`, because there are no frames at all.